prelude.dx

```diff
@@ -697,20 +704,31 @@ def evalpoly (_:VSpace v) ?=> (coefficients:n=>v) (x:Float) : v =
   -- Evaluate a polynomial at x.  Same as Numpy's polyval.
   fold zero \i c. coefficients.i + x .* c
 
+
+'Monoid
+
+interface Monoid a:Type where
+  mempty : a
+  mcombine : a -> a -> a  -- can't use `<>` just for parser reasons?
```

Comment:
This is probably because `interface` defines the ADT in terms of a record, and `(<>)` isn't a valid record field.

Reply:
Not sure. It might actually be fine to just use `(<>)` as the key in the record even though users can never write that as a record key. I don't think users will be manually manipulating interface dicts anyways.

Reply:
Alternatively, we can simply mangle the interface function name when putting them into records. As you've said, users will never be manipulating those by hand, so we're free to rewrite them as we'd like.